packages/clerk-js/src/core/resources/Session.ts

```diff
@@ -151,28 +151,38 @@ export class Session extends BaseResource implements SessionResource {
       return null;
     }
 
-    const { leewayInSeconds, template, skipCache = false } = options || {};
+    const {
+      leewayInSeconds,
+      template,
+      skipCache = false,
+      organizationId = Session.clerk.organization?.id,
+    } = options || {};
+
     if (!template && Number(leewayInSeconds) >= 60) {
       throw new Error('Leeway can not exceed the token lifespan (60 seconds)');
     }
 
-    const tokenId = this.#getCacheId(template);
+    const tokenId = this.#getCacheId(template, organizationId);
     const cachedEntry = skipCache ? undefined : SessionTokenCache.get({ tokenId }, leewayInSeconds);
 
+    // Dispatch tokenUpdate only for __session tokens with the session's active organization ID, and not JWT templates
+    const shouldDispatchTokenUpdate = !template && options?.organizationId === Session.clerk.organization?.id;
```

Comment:
🤔 I am kind of sceptic about using the ` Session.clerk.organization?.id` here :/.
Also why do we enforce that the token:update event should be dispatched only for the same organization id?

Reply:
@dimkl We don't want the session cookie to contain an org ID that isn't the active one. So if someone provides a different org ID that does not match the active org for the session, we don't want the cookie to be updated.

Reply:
Do you have a recommendation for a better way to access the currently selected org?

Reply:
Could we take it from the session claims or from the `this.lastActiveOrganizationId`?
Also have we checked that the switch organization scenario updates the session cookie correctly or is the `token:update` event skipped and the cookie stays the same?

Reply:
switching the organization via `setActive()` dispatches `TokenUpdate` separately, and I confirmed it updates the cookie as expected!